src/bots/bot.ts

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable no-warning-comments */
+import { IBot } from '../types'
+
+// 1 minutes = 60_000 milliseconds
+const ONE_MINUTE = 60_000
+
+export class Bot implements IBot {
+  username: string
+  password: string
+  private hitCount = 0
+  private loginCount = 0
+  private captchaCount = 0
+  // rateLimit of 0 means no rate limiting
+  // eslint-disable-next-line @typescript-eslint/prefer-readonly
+  private rateLimit = 0
```

Comment:
why not follow the eslint suggestion here?

Reply:
thought we might want to change rate-limit after instantiating as well.